File: solutions/search-insert-position/search-insert-position.cpp
// @Title: Search Insert Position
// @Author: colinjxc
// @Date: 2022-02-08T03:44:31+08:00
// @URL: https://leetcode-cn.com/problems/search-insert-position


class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int j = 0, k = nums.size();
        while (j < k) {
            int m = j + (k -j )/2;
            if (nums[m] < target) {
                j = m + 1;
            } else  {
                k = m; 
            }
        }
        return k;
    }
};

File: solutions/search-insert-position/search-insert-position.py
// @Title: Search Insert Position
// @Author: colinjxc
// @Date: 2022-01-26T01:07:07+08:00
// @URL: https://leetcode-cn.com/problems/search-insert-position


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums)
        while i < j:
            m = (i + j)//2
            if nums[m] < target:
                i = m + 1
            else:
                j = m
        
        return i
